### PR TITLE
feat: Improve search support from SDK

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

- Add filter support for `id` and `external_id`
- Add filter by `inserted_at` and `updated_at`
- Add filter support for other suggestions and response attributes (~~user_id~~, agent, type,...)
- Add similarity search support - https://github.com/argilla-io/argilla/pull/5546

**Type of change**

- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by jfcalvo — APPROVED** [review]

**Review by davidberenstein1957 — COMMENTED** [review]
LFTM, I think we still need to add some references here. https://docs.argilla.io/latest/how_to_guides/query/

**Comment by frascuchon** [comment]
> LFTM, I think we still need to add some references here. https://docs.argilla.io/latest/how_to_guides/query/

Done

File: argilla/docs/how_to_guides/query.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
----
+from huggingface_hub import dataset_infofrom tests.integration.conftest import client---
```
Comment (davidberenstein1957) [inline argilla/docs/how_to_guides/query.md]: why was this added?

File: argilla/docs/how_to_guides/query.md [hunk 2]
```diff
@@ -31,6 +31,16 @@ You can search for records in your dataset by **querying** or **filtering**. The
...
         > Check the [Filter - Python Reference](../reference/argilla/search.md) to see the attributes, arguments, and methods of the `Filter` class in detail.
 
+    === "`rg.Similar`"
+
+        ```python
+        rg.Similar(
+            name="vector",
+            value=(0.1, 0.2, 0.3),
+        )
+        ```
```
Comment (davidberenstein1957) [inline argilla/docs/how_to_guides/query.md]: Perhaps show a list here.

File: argilla/docs/how_to_guides/query.md [hunk 3]
```diff
@@ -153,6 +183,32 @@ status_filter = rg.Query(
...
+import argilla as rg
+
+client = rg.Argilla(api_url="<api_url>", api_key="<api_key>")
+
+dataset = client.datasets(name="my_dataset", workspace="my_workspace")
+
+
+similar_filter = rg.Query(
+    similar=rg.Similar(
+        name="vector", value=(0.1, 0.2, 0.3),
```
Comment (davidberenstein1957) [inline argilla/docs/how_to_guides/query.md]: why a tuple?

File: argilla/docs/how_to_guides/query.md (hunk 1 above)
Comment (frascuchon) [reply argilla/docs/how_to_guides/query.md]: thanks to copilot I guess

Comment (frascuchon) [inline argilla/docs/how_to_guides/query.md]:
```suggestion
```

File: argilla/docs/how_to_guides/query.md [hunk 4]
```diff
@@ -31,6 +31,16 @@ You can search for records in your dataset by **querying** or **filtering**. The
         ```
         > Check the [Filter - Python Reference](../reference/argilla/search.md) to see the attributes, arguments, and methods of the `Filter` class in detail.
 
+    === "`rg.Similar`"
+
+        ```python
+        rg.Similar(
+            name="vector",
+            value=(0.1, 0.2, 0.3),
```
Comment (frascuchon) [inline argilla/docs/how_to_guides/query.md]:
```suggestion
            value=[0.1, 0.2, 0.3],
```

File: argilla/docs/how_to_guides/query.md (hunk 3 above)
Comment (frascuchon) [inline argilla/docs/how_to_guides/query.md]:
```suggestion
        name="vector", value=[0.1, 0.2, 0.3],
```

File: argilla/docs/how_to_guides/query.md (hunk 4 above)
Comment (davidberenstein1957) [inline argilla/docs/how_to_guides/query.md]:
```suggestion
            value=[0.1, 0.2, 0.3],
```

File: argilla/src/argilla/_models/_search.py [hunk 5]
```diff
@@ -60,9 +61,21 @@ class TermsFilterModel(BaseModel):
...
     scope: ScopeModel
 
+    @field_serializer("values", when_used="unless-none")
+    def serialize_values(self, values):
+        sanitized_values = []
+        for value in values:
+            if isinstance(value, UUID):
+                value = str(value)
+            elif isinstance(value, datetime):
+                value = value.isoformat()
```
Comment (davidberenstein1957) [inline argilla/src/argilla/_models/_search.py:74]: Nice!

**Comment by davidberenstein1957** [comment]
@frascuchon final remark w.r.t. the docs.

File: argilla/docs/how_to_guides/query.md (hunk 1 above)
Comment (frascuchon) [inline argilla/docs/how_to_guides/query.md]:
```suggestion
---
```

**Review by davidberenstein1957 — APPROVED** [review]
